internal/rbd/rbd_journal.go

```diff
@@ -598,16 +602,27 @@ func RegenerateJournal(
 	if err != nil {
 		return "", err
 	}
+
 	if imageData != nil {
 		rbdVol.ReservedID = imageData.ImageUUID
 		rbdVol.ImageID = imageData.ImageAttributes.ImageID
 		rbdVol.Owner = imageData.ImageAttributes.Owner
+		rbdVol.RbdImageName = imageData.ImageAttributes.ImageName
 		if rbdVol.ImageID == "" {
 			err = rbdVol.storeImageID(ctx, j)
 			if err != nil {
 				return "", err
 			}
 		}
+		// Update Metadata on reattach of the same old PV
+		parameters := map[string]string{
+			pvcNameKey:      claimName,
```

Comment:
find a way to use a helper function under the `internal/utils/k8s` package to create the `parameters` map